b2share/modules/communities/api.py

```diff
@@ -18,73 +18,237 @@
+
+        Returns:
+            :class:`Community`: self
+
+        Raises:
+            b2share.modules.communities.errors.InvalidCommunityError: The
+                community update failed because the resulting community is
+                not valid.
+        """
+        try:
+            with db.session.begin_nested():
+                before_community_update.send(self)
+                if clear_fields:
+                    for field in ['name', 'description', 'logo']:
+                        setattr(self.model, field, data.get(field, None))
+                else:
+                    for key, value in data.items():
+                        setattr(self.model, key, value)
+                db.session.merge(self.model)
+        except sqlalchemy.exc.IntegrityError as e:
+            raise InvalidCommunityError() from e
+        after_community_update.send(self)
+        return self
+
+    def patch(self, patch):
+        """Update the community's metadata with a json-patch.
+
+        Args:
+            patch (str): json-patch which can modify the following fields:
+                name, description, logo.
+
+        Returns:
+            :class:`Community`: self
+
+        Raises:
+            jsonpatch.JsonPatchConflict: the json patch conflicts on the
+                community.
+
+            jsonpatch.InvalidJsonPatch: the json patch is invalid.
+
+            b2share.modules.communities.errors.InvalidCommunityError: The
+                community patch failed because the resulting community is
+                not valid.
+        """
+        data = apply_patch({
+            'name': self.model.name,
+            'description': self.model.description,
+            'logo': self.model.logo
+        }, patch, True)
+        self.update(data)
+        return self
+
+    # TODO: add properties for getting schemas and admins
+
+    def delete(self):
+        """Mark a community as deleted.
+
+        Returns:
+            :class:`Community`: self
+        """
+        with db.session.begin_nested():
+            before_community_delete.send(self)
+            self.model.deleted = True
+            db.session.merge(self.model)
+            # FIXME: What do we do with this community's records?
+        after_community_delete.send(self)
+        return self
+
+    @property
+    def id(self):
+        """Get community id."""
+        return self.model.id
+
+    @property
+    def created(self):
+        """Get creation timestamp."""
+        return self.model.created
+
+    @property
+    def updated(self):
+        """Get last updated timestamp."""
+        return self.model.updated
+
+    @property
+    def deleted(self):
+        """Get community deletion state.
+
+        Returns:
+            date: True if the community is deleted, else False.
+        """
+        return self.model.deleted
+
+    @property
+    def name(self):
+        """Retrieve community's name."""
+        return self.model.name
+
+    @name.setter
```

Comment:
My bad, I was sure I had removed them. I must have reverted the change during the rebases.